### DISCUSSION TITLE
Welcome to zhang Discussions!

### DISCUSSION THREAD

**Discussion by Kilerd** [Kilerd/zhang#153] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Kilerd** [comment]
1. 关键字应该是可以搜索 payee 和 narration 的
2. 每页数量这个可以考虑增加进去，目前是固定的
3.点击页数无反应估计是bug
4. 输入页数，这个功能会很常用吗？ 
5. 没有折线图的话，是不是这个帐户在这段时间里面没有交易波动？ 看你的最后一张截图似乎图表展示出来了，但是金额没有波动

**Reply by Kilerd** [reply]
I've created the issue for the pagination https://github.com/zhang-accounting/zhang/issues/389

and I'll impl it in days

**Reply by sea-ten** [reply]
> 1. 关键字应该是可以搜索 payee 和 narration 的
> 2. 每页数量这个可以考虑增加进去，目前是固定的
>    3.点击页数无反应估计是bug
> 3. 输入页数，这个功能会很常用吗？
> 4. 没有折线图的话，是不是这个帐户在这段时间里面没有交易波动？ 看你的最后一张截图似乎图表展示出来了，但是金额没有波动

1. 输入页数这个功能其实是针对过滤结果很多的情况下需要跳转到某一年某一月的时使用的，不过如果过滤条件支持关键字和时间，就不需要输入页面这个功能了。
2. 折线图问题，该账户近段时间内的波动是比较小，但是从全时间段来说，是有起伏的，不过我发现这个没折线的账户，在前面有几条余额为负数的记录，不知道是不是这个原因。我昨天为了解决balance问题修改了初始化时间，我重新调整一下看看有没有变化。

**Reply by Kilerd** [reply]
折线图这个异常，应该就是负数导致的。 我这个折线图采用了指数型图表，可以在大金额的情况下，通过指数显示的模式来放大细微的改动。 可是这对负数不生效，可能得加一个判断，如果存在负数就得用回普通图表。

**Reply by sea-ten** [reply]
是的，我调整了一下，现在折线图恢复正常了，确实是余额有负数导致的。

**Comment by Kilerd** [comment]
抱歉，你说的搜索没反应是这样的

在journal页面有两个搜索框
<img width="1379" alt="image" src="https://github.com/user-attachments/assets/5bdc98ef-f538-4b25-b82c-97e11611d61c">

目前只有左边下面这个是有效的， 右上角的搜索框现在只是一个占位，还没有实现功能 hhhh

**Reply by sea-ten** [reply]
哈哈，我刚截图完毕，不知道你有什么设计规划，我觉得可以用来过滤时间(段)，双重条件约束下，过滤结果会更精确。
![image](https://github.com/user-attachments/assets/f6514a79-1e38-4c21-8f66-0790ec0b73e9)

**Reply by Kilerd** [reply]
我个人倾向于它是作为一个当前页面的检索入口，在流水页面执行的是流水检索，在帐户列表页面执行帐户检索。 不过可能会有点重复了。

至于你说的时间过滤，应该可以在左下角的检索框边上加一个时间range的条件

**Reply by sea-ten** [reply]
这样子更好👍👍

**Comment by sea-ten** [comment]
对了，我有个新问题想请教一下，比如说我通过关键字搜索出来我在KFC的所有消费记录，但我现在无法直观的看到某个时间段在KFC的消费总额，这个功能后面会实现吗？

**Reply by Kilerd** [reply]
我在构思类似于beancount sql的复杂查询工具，然后可以支持把结果生成图表，以满足每个人的复杂、个性化查询。

**Reply by sea-ten** [reply]
太感谢了
